runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -104,11 +108,31 @@ public void start() throws IOException {
             log.error("failed to run restore.", e);
             throw new BackupRestoreException("failed to restore from backup file " + filePath, e);
         } finally {
+            enableCompaction();
             cleanup();
         }
         log.info("restore completed");
     }
 
+    private void disableCompaction() throws Exception {
+        log.info("Disabling compaction...");
+        if (cpHelper == null) {
+            try {
+                cpHelper = new DistributedCheckpointerHelper(corfuStore);
+            } catch (Exception e) {
+                log.error("Failed to obtain a DistributedCheckpointerHelper.", e);
+                throw e;
+            }
+        }
+
+        cpHelper.disableCompaction();
```

Comment:
This should wait for some time longer than the compactor orchestrator scheduling period so that it ensures the disable is effective on all three nodes.